### PR TITLE
Refactor and Fix hosters and clientarea

### PR DESCRIPTION
**Pull request by Erackron**
This PR includes:
- Documentation updates (dependencies for unit tests should be installed by installing the package with extra_requires: [test], so for example: `pip install cloudomate[test]`)
- Everything is now Python 2 and 3 compatible
- Status and Info for all VPS and VPN providers is tested and functional
- Removal of Legionbox due to phone number verification
- fake-useragent dependency for radom useragent
- Unified root password change by using ssh
- Explicitly implemented the bitcoin payment provider used by Underground Private (it was already used, but not specified properly yet)

Ideally I would have wanted to create multiple commits, but these changes were pretty hard to separate since they had to be rebased on this branch after they were based on the previous (unsquashed and unedited) PR.

### CODE REVIEW

**Comment by devos50** [comment]
ok to test

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

File: cloudomate/cmdline.py [hunk 1]
```diff
@@ -330,42 +397,76 @@ def _print_unknown_provider_type(provider_type):
...
+        bandwidth = "Unlimited" if option.bandwidth == sys.maxsize else str(option.bandwidth)
+
+        # Calculate the estimated price
+        rate = wallet_util.get_rate("USD")
+        fee = wallet_util.get_network_fee()
+        gateway = p.get_gateway()
+        estimate = gateway.estimate_price(option.price * rate) + fee  # BTC
+        estimate = round(1000 * estimate, 2)  # mBTC
+
+        # Print everything
```
Comment (devos50) [inline cloudomate/cmdline.py]: This comment is a bit unnecessary.

File: cloudomate/cmdline.py [hunk 2]
```diff
@@ -330,42 +397,76 @@ def _print_unknown_provider_type(provider_type):
...
+        speed = "Unlimited" if option.speed == sys.maxsize else option.speed
+
+        # Calculate the estimated price
+        rate = wallet_util.get_rate("USD")
+        fee = wallet_util.get_network_fee()
+        gateway = provider.get_gateway()
+        estimate = gateway.estimate_price(option.price * rate) + fee  # BTC
+        estimate = round(1000 * estimate, 2)  # mBTC
+
+        # Print everything
```
Comment (devos50) [inline cloudomate/cmdline.py]: See previous comment.

File: cloudomate/cmdline.py [hunk 3]
```diff
@@ -330,42 +397,76 @@ def _print_unknown_provider_type(provider_type):
...
     # If wallet path is defined in config, use that.
-    if 'walletpath' in user_settings.config:
-        wallet = Wallet(wallet_path=user_settings.get('walletpath'))
+    if settings.has_key('client', 'walletpath'):
+        wallet = Wallet(wallet_path=settings.get('client', 'walletpath'))
     else:
         wallet = Wallet()
 
-    p.purchase(user_settings=user_settings, wallet=wallet)
+    provider = p(settings)
```
Comment (devos50) [inline cloudomate/cmdline.py]: Is this `p` a class? Please rename it to something more meaningful.

File: cloudomate/hoster/vps/clientarea.py [hunk 4]
```diff
@@ -1,256 +1,110 @@
...
-        If number is set through config or cli, return the service according to the number, else return the first 
-        service.
-        :return: the chosen service
-        """
-        number = self._get_number()
-        self._services()
-        self._verify_number(number)
-        return self.services[number]
+        price = columns[1].text
+        i = price.index('.')
```
Comment (devos50) [inline cloudomate/hoster/vps/clientarea.py]: Please rename these variables (`i`, `p`,  `c` etc) to something more meaningful.

File: cloudomate/hoster/vps/crowncloud.py [hunk 5]
```diff
@@ -1,171 +1,163 @@
...
+    Hoster-specific methods that are needed to perform the actions
+    '''
+
+    @classmethod
+    def _parse_options(cls, page):
+        tables = page.findAll('table')
+        for table in tables:  # There are multiple tables with server options on the page
+            for row in table.findAll('tr'):
+                if len(row.findAll('td')) > 0:  # Ignore headers
+                    r = cls._parse_row(row)
```
Comment (devos50) [inline cloudomate/hoster/vps/crowncloud.py]: Unclear variable name.

File: cloudomate/hoster/vps/undergroundprivate.py [hunk 6]
```diff
@@ -1,139 +1,138 @@
...
+
+        cores = details[2].text
+        cores = cores.split('\n')
+        cores = int(cores[1][0])
+
+        memory = details[4].text
+        memory = float(memory[0])
+
+        storage = details[3].text
+        i = storage.index('GB')
```
Comment (devos50) [inline cloudomate/hoster/vps/undergroundprivate.py]: Unclear variable name.

**Comment by Erackron** [comment]
@devos50 Requested changes have been made

**Pull request merged**